crates/torii/server/src/cli.rs

```diff
@@ -154,3 +121,73 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+// Tries to find scarb manifest first for env variables
+//
+// Use manifest path from cli args,
+// else uses scarb manifest to derive path of dojo manifest file,
+// else try to derive manifest path from scarb manifest
+// else try `./target/dev/manifest.json` as dojo manifest path
+//
+// If neither of this work return an error and exit
+fn get_manifest_and_env(
+    args_path: Option<&Utf8PathBuf>,
+) -> anyhow::Result<(Manifest, Option<Environment>)> {
+    // If there is scarb manifest file it shoul
+    let config;
+    let ws = if let Ok(scarb_manifest_path) = scarb::ops::find_manifest_path(None) {
+        config = Config::builder(scarb_manifest_path)
+            .log_filter_directive(env::var_os("SCARB_LOG"))
+            .build()
+            .with_context(|| "Couldn't build scarb config".to_string())?;
+        scarb::ops::read_workspace(config.manifest_path(), &config).ok()
+    } else {
+        None
+    };
+
+    let manifest = if let Some(manifest_path) = args_path {
+        Manifest::load_from_path(manifest_path)?
+    } else if ws.is_some() {
+        let ws = ws.as_ref().unwrap();
+        let target_dir = ws.target_dir().path_existent()?;
+        let target_dir = target_dir.join(ws.config().profile().as_str());
+        let manifest_path = target_dir.join("manifest.json");
+        Manifest::load_from_path(manifest_path)?
+    } else {
+        Manifest::load_from_path(Utf8PathBuf::from("./target/dev/manifest.json"))?
+    };
```

Comment:
Whats the reasoning for defaulting to `target/dev/manifest.json` here?

To assume that `target` dir exists would also to assume that we're already in a `scarb` workspace. Don't think there'd be any situation where `target/dev/` would exists without a `scarb` workspace in the context of Dojo.

 Would be redundant when we already checking if `scarb` exists.

Reply:
make sense, will remove